src/uu/touch/src/touch.rs

```diff
@@ -339,8 +341,14 @@ fn parse_date(ref_time: DateTime<Local>, s: &str) -> UResult<FileTime> {
     // Tue Dec  3 ...
     // ("%c", POSIX_LOCALE_FORMAT),
     //
-    if let Ok(parsed) = NaiveDateTime::parse_from_str(s, format::POSIX_LOCALE) {
-        return Ok(datetime_to_filetime(&parsed.and_utc()));
+    if let Ok((parsed, modifier)) = NaiveDateTime::parse_and_remainder(s, format::POSIX_LOCALE) {
+        if modifier.is_empty() {
+            return Ok(datetime_to_filetime(&parsed.and_utc()));
+        }
+        return match date_from_modifier(modifier, parsed) {
+            Ok(new_date) => Ok(datetime_to_filetime(&new_date.and_utc())),
+            Err(e) => Err(e),
+        };
```

Comment:
I think it could be simplified with:
```suggestion
return date_from_modifier(modifier, parsed).map(|new_date| datetime_to_filetime(&new_date.and_utc()));
```